A table and routes for supplier contracts - API for #803

flypunk opened the pull request:
This branch provides GET, POST, and DELETE verbs for /suppliers/:supplier_id/contract route.
POST supports both uploading a new contract and replacing one in place.

In order to test it, you will have to run `knex migrate:latest` and have the following environment variables in your .env file:

SPARK_SUPPLIER_CONTRACTS_BUCKET
AWS_SECRET_ACCESS_KEY
AWS_ACCESS_KEY_ID

The SPARK_SUPPLIER_CONTRACTS_BUCKET has to exist beforehand and be accessible by the
owner of the AWS keys.

flypunk committed:
Supplier contracts - WIP

flypunk committed:
Drop unneeded column

flypunk committed:
Add SupplierContract model

flypunk committed:
Add a bucket for supplier contracts

flypunk committed:
Add checking for existing supplier; Insert or update the contract in the DB

flypunk committed:
Add a route for GETting supplier contract (#803)

flypunk committed:
Add a route for DELETing supplier contract (#803)

OriHoch reviewed (approved):
:+1:

flypunk committed:
Resolving rebase conflict

flypunk committed:
Resolving rebase conflict

guyzohar1 commented on migrations/20180327173047_add_supplier_contracts.js:
```diff
@@ -0,0 +1,18 @@
+        // Suppliers table
+        knex.schema.createTable(constants.SUPPLIERS_CONTRACTS_TABLE_NAME, function(table) {
+            table.timestamps();
+            table.integer('supplier_id', 9).unsigned().primary();
```
change to string

flypunk replied:
👍

guyzohar1 commented on migrations/20180327173047_add_supplier_contracts.js:
```diff
@@ -0,0 +1,18 @@
+        knex.schema.createTable(constants.SUPPLIERS_CONTRACTS_TABLE_NAME, function(table) {
+            table.timestamps();
+            table.integer('supplier_id', 9).unsigned().primary();
+            table.string('contract_name', 100);
```
i think this should be called path or url

flypunk replied:
I am changing `contract_name` to `file_name`, because it's just the leaf name and not a full path.
This string is also used as a key for S3 bucket.

flypunk committed:
Improvements - supplier contract management (#803)

- Change the type of supplier_id column to string
- Change the column name of contract file to `file_name`

flypunk committed:
Improvements - supplier contract management (#803)

- Add routes documentation
- Limit access to camp admins
- Change a contract column name to `file_name`

flypunk committed:
Merge branch 'suppliers' into supplier-documents

flypunk committed:
Delete extraneous userRole declaration (#803)
